### PR TITLE
[Storage] Switch from `ClientSecretCredential` to `DefaultAzureCredential`

### PR DESCRIPTION
**Pull request by vincenttran-msft**
This PR aims to move away from using discrete `ClientSecretCredential` for OAuth operations and to instead now use `DefaultAzureCredential`.

This also means that you have to routinely do `az login` when running OAuth tests to refresh your permission, but will get away from those pesky secrets 😄 

TODO:

- Get this passing in CI 
- Get this passing in live tests

### CODE REVIEW

File: tools/azure-sdk-tools/devtools_testutils/envvariable_loader.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def create_resource(self, name, **kwargs):
         load_dotenv(find_dotenv())
 
         if self.is_live:
-            self._set_mgmt_settings_real_values()
+            # self._set_mgmt_settings_real_values()
```
Comment (vincenttran-msft) [inline tools/azure-sdk-tools/devtools_testutils/envvariable_loader.py:99]: Commenting this out for now--

But if we still need this functionality to be supported, we will need some sort of logic here to conditionally call that helper function (or to simply pass if not set and maybe raise an error)?

**Review by mccoyp — APPROVED** [review]
LGTM! We'll keep an eye out for any usability quirks and make things more Storage setup-compatible if necessary.

**Comment by vincenttran-msft** [comment]
Closing this to avoid confusion -- this was an attempt to work with the currently infra but ended up going a whole refactor and was a collaborative effort 😄

**Pull request closed without merging**